ThirdPartyAdapters/applovin/applovin/src/main/java/com/applovin/mediation/AppLovinNativeAdMapper.java

```diff
@@ -0,0 +1,96 @@
+package com.applovin.mediation;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.applovin.nativeAds.AppLovinNativeAd;
+import com.google.android.gms.ads.formats.NativeAd;
+import com.google.android.gms.ads.mediation.NativeAppInstallAdMapper;
+
+import java.util.ArrayList;
+
+class AppLovinNativeAdMapper extends NativeAppInstallAdMapper {
+
+    private AppLovinNativeAd mNativeAd;
+
+    AppLovinNativeAdMapper(AppLovinNativeAd nativeAd, Context context) {
+        mNativeAd = nativeAd;
+        setHeadline(nativeAd.getTitle());
+        setBody(nativeAd.getDescriptionText());
+        setCallToAction(nativeAd.getCtaText());
+
+        ImageView mediaView = new ImageView(context);
+        ViewGroup.LayoutParams layoutParams =
+                new ViewGroup.LayoutParams(
+                        ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+        mediaView.setLayoutParams(layoutParams);
+
+        ArrayList<NativeAd.Image> images = new ArrayList<>(1);
+        Drawable imageDrawable = Drawable.createFromPath(Uri.parse(nativeAd.getImageUrl()).getPath());
+        Drawable iconDrawable = Drawable.createFromPath(Uri.parse(nativeAd.getIconUrl()).getPath());
+        AppLovinNativeAdImage image = new AppLovinNativeAdImage(Uri.parse(nativeAd.getImageUrl()), imageDrawable);
+        AppLovinNativeAdImage icon = new AppLovinNativeAdImage(Uri.parse(nativeAd.getIconUrl()), iconDrawable);
+        images.add(image);
+        setImages(images);
+        setIcon(icon);
+
+        mediaView.setImageDrawable(imageDrawable);
+        setMediaView(mediaView);
+        setStarRating(nativeAd.getStarRating());
+
+        Bundle extraAssets = new Bundle();
```

Comment:
The point of `extraAssets`? Is it being used anywhere?

Reply:
Any assets that cannot be directly mapped to an AdMob ad can be passed to the publisher using the extras bundle.

The publisher would use the keys in `AppLovinNativeAdapter` class to access the assets from the bundle. These keys need to be documented (See [Facebook adapter documentation](https://developers.google.com/admob/android/mediation/facebook#using_native_ads) for example)

Reply:
1. I still don't get how the publisher can access this `Bundle`. It looks like you're just creating a `Bundle`, adding the ad id and caption into it, and it'll get deallocated, right? Are you missing a superclass call like a `setBundleExtras(xxx)` or something?
2. How come you're not passing in the video? Are we going to add support for videos in this release, or in a future one?

Reply:
Ah, I missed the `setExtras(Bundle)`call. Added it now. The publisher would be able to now get them using the [`getExtras()`](https://developers.google.com/android/reference/com/google/android/gms/ads/formats/NativeAppInstallAd.html#getExtras()) method on `NativeAppInstallAd`.